src/EditorFeatures/CSharpTest/ChangeSignature/AddParameterTests.Cascading.cs

```diff
@@ -0,0 +1,412 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ChangeSignature;
+using Microsoft.CodeAnalysis.Editor.UnitTests.ChangeSignature;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities.ChangeSignature;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.ChangeSignature
+{
+    public partial class ChangeSignatureTests : AbstractChangeSignatureTests
+    {
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ChangeSignature)]
+        public async Task AddParameter_Cascade_ToImplementedMethod()
+        {
+            var markup = @"
+interface I
+{
+    void M(int x, string y);
+}
+
+class C : I
+{
+    $$public void M(int x, string y)
+    { }
+}";
+            var permutation = new[] {
+                new AddedParameterOrExistingIndex(1),
+                new AddedParameterOrExistingIndex(new AddedParameter(null, "int", "newIntegerParameter", "12345"), "int"),
+                new AddedParameterOrExistingIndex(0)
```

Comment:
can these not have implicit conversions to `AddedParameterOrExistingIndex`?  then this could be `new AddedParameterOrExistingIndex[] { 1, new AddedParameter(...), 0 }`.

Also, in tests, consider having helper methods o avoid `new` everywhere.  i.e. `Added(null, ...)` which produces either an AddedParameter or an AddedParameterOrIndex.

Reply:
why is `int` duplicated?

Reply:
Filed https://github.com/dotnet/roslyn/issues/42734 to reorganize & clean up test helpers.